src/openvic-simulation/history/ProvinceHistory.cpp

```diff
@@ -15,6 +16,37 @@ std::unique_ptr<ProvinceHistoryEntry> ProvinceHistoryMap::_make_entry(Date date)
 	return std::unique_ptr<ProvinceHistoryEntry> { new ProvinceHistoryEntry { province, date } };
 }
 
+std::unique_ptr<ProvinceHistoryEntry> ProvinceHistoryMap::_get_entry(Date date) {
+	return std::move(entries.at(date));
+}
```

Comment:
Why is this returning a unique_ptr and not a pointer?

Reply:
Because `entries` is a map of `Date` to `unique_ptr`, and I had no idea of how to unwrap it.

Reply:
[`std::unique_ptr::operator*`](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*) and [`std::unique_ptr::get`](https://en.cppreference.com/w/cpp/memory/unique_ptr/get)

Returning a unique_ptr here is bad because moving a unique_ptr swaps the data with a nullptr, so the pointer entry in `entries` will become null. Every case for which you move a unique_ptr does this, never move a unique_ptr unless you are moving the ownership of the data, as in the responsibility for freeing the pointer data. This is what dereferencing the unique_ptr is for.

Reply:
I understand, I'll fix it right now.